modules/charon-core/src/main/java/org/wso2/charon3/core/config/ExtensionBuilder.java

```diff
@@ -0,0 +1,495 @@
+            type = getDefinedDataType(attributeConfig.get(SCIMConfigConstants.DATA_TYPE));
+            multiValued = Boolean.valueOf(attributeConfig.get(SCIMConfigConstants.MULTIVALUED));
+            description = attributeConfig.get(SCIMConfigConstants.DESCRIPTION);
+            required = Boolean.parseBoolean(attributeConfig.get(SCIMConfigConstants.REQUIRED));
+            caseExact = Boolean.parseBoolean(attributeConfig.get(SCIMConfigConstants.CASE_EXACT));
+            mutability = getDefinedMutability(attributeConfig.get(SCIMConfigConstants.MUTABILITY));
+            returned = getDefinedReturned(attributeConfig.get(SCIMConfigConstants.RETURNED));
+            uniqueness = getDefinedUniqueness(attributeConfig.get(SCIMConfigConstants.UNIQUENESS));
+            String subAttributesString = attributeConfig.get(SCIMConfigConstants.SUB_ATTRIBUTES);
+            if (StringUtils.isNotBlank(subAttributesString) && !"null".equalsIgnoreCase(subAttributesString)) {
+                subAttributes = subAttributesString.split(" ");
+            }
+        }
+
+        public ExtensionAttributeSchemaConfig(JSONObject attributeConfigJSON) throws CharonException {
+
+            try {
+                uri = attributeConfigJSON.getString(SCIMConfigConstants.ATTRIBUTE_URI);
+                name = attributeConfigJSON.getString(SCIMConfigConstants.ATTRIBUTE_NAME);
+                type = getDefinedDataType(attributeConfigJSON.getString(SCIMConfigConstants.DATA_TYPE));
+                multiValued = attributeConfigJSON.getBoolean(SCIMConfigConstants.MULTIVALUED);
+                description = attributeConfigJSON.getString(SCIMConfigConstants.DESCRIPTION);
+                required = Boolean.parseBoolean(attributeConfigJSON.getString(SCIMConfigConstants.REQUIRED));
+                caseExact = Boolean.parseBoolean(attributeConfigJSON.getString(SCIMConfigConstants.CASE_EXACT));
+                mutability = getDefinedMutability(attributeConfigJSON.getString(SCIMConfigConstants.MUTABILITY));
+                returned = getDefinedReturned(attributeConfigJSON.getString(SCIMConfigConstants.RETURNED));
+                uniqueness = getDefinedUniqueness(attributeConfigJSON.getString(SCIMConfigConstants.UNIQUENESS));
+                String subAttributesString = attributeConfigJSON.getString(SCIMConfigConstants.SUB_ATTRIBUTES);
+                if (!"null".equalsIgnoreCase(subAttributesString)) {
+                    subAttributes = subAttributesString.split(" ");
+                }
+                canonicalValues =
+                        setCanonicalValues(attributeConfigJSON.getJSONArray(SCIMConfigConstants.CANONICAL_VALUES));
+                referenceTypes =
+                        setReferenceTypes(attributeConfigJSON.getJSONArray(SCIMConfigConstants.REFERENCE_TYPES));
+            } catch (JSONException e) {
+                throw new CharonException("Error while parsing extension configuration", e);
+            }
+        }
+
+        /*
```

Comment:
Should be a doc comment.

Reply:
There are multiple instances. Fix all.

Reply:
fixed all with https://github.com/wso2/charon/pull/323/commits/6a773e3384a5db3baab94370f9f64a904834a48d